### PR TITLE
Github Actions: Migrated base to ubuntu-latest & Grouped Sanity Tests

### LINKED ISSUE #23498: GithubActions - Use ubuntu-latest as base
#### Is your feature request related to a problem?

After #23490, we move from Ubuntu 20.04 to Ubuntu 18.04

#### Describe the solution you'd like

We should use latest version of Github Actions for avoiding technical debt.

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Github Actions: Migrated base to ubuntu-latest & Grouped Sanity Tests & :heart: on @atomiix 
| Type?             | improvement
| Category?         | PM
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #23498
| How to test?      | Github Actions are :green_circle: 

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/25288)

### CODE REVIEW

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ on: [push, pull_request]
 jobs:
     integration:
         name: Integration tests
-        runs-on: ubuntu-18.04
+        runs-on: ubuntu-latest
```
Comment (PierreRambaud) [inline .github/workflows/integration.yml:6]: You have to download mysqldump tool when running on latest Ubuntu release ;)

Comment (Progi1984) [reply .github/workflows/integration.yml:6]: Oh... Thanks ;)

Comment (Progi1984) [reply .github/workflows/integration.yml:6]: That doesn't seem the only need

Comment (PierreRambaud) [reply .github/workflows/integration.yml:6]: Maybe some dependencies are missing, it's time to debug it :sweat_smile:

**Review by PierreRambaud — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Comment by PierreRambaud** [comment]
Thank you @Progi1984 and @atomiix

**Pull request merged**